### ISSUE TITLE
关于训练完成后模型效果的提问

### ISSUE THREAD

**Issue by cutecharmingkid** [PeiChangY/-BERT-BILSTM-DGCNN-ATTENTION-CRF-#3]
您好，我是使用您的数据集msra_data进行训练，我使用predict文件预测使用的语句是：习近平是中华人民共和国的主席。返回的预测结果为空。同时系统提示：UserWarning: where received a uint8 condition tensor. This behavior is deprecated and will be removed in a future version of PyTorch. Use a boolean condition instead. (Triggered internally at ..\aten\src\ATen\native\TensorCompare.cpp:328.)
score = torch.where(mask[i].unsqueeze(1), next_score, score)
![887ea71b0140ffbaf084f56d77b478b](https://user-images.githubusercontent.com/102043624/225259519-c6a83f71-6601-42e6-b459-61c38851695a.png)

**Comment by PeiChangY** [comment]
建议您看一下我发表在知网上的论文《基于门控空洞卷积神经网络的中文命名实体识别》，里面讲的比较详细，有各个数据集的实体类型




------------------&nbsp;原始邮件&nbsp;------------------
发件人:                                                                                                                        "PeiChangY/-BERT-BILSTM-DGCNN-ATTENTION-CRF-"                                                                                    ***@***.***&gt;;
发送时间:&nbsp;2023年3月15日(星期三) 下午5:03
***@***.***&gt;;
***@***.***&gt;;
主题:&nbsp;[PeiChangY/-BERT-BILSTM-DGCNN-ATTENTION-CRF-] 关于训练完成后模型效果的提问 (Issue #3)





 
您好，我是使用您的数据集msra_data进行训练，我使用predict文件预测使用的语句是：习近平是中华人民共和国的主席。返回的预测结果为空。同时系统提示：UserWarning: where received a uint8 condition tensor. This behavior is deprecated and will be removed in a future version of PyTorch. Use a boolean condition instead. (Triggered internally at ..\aten\src\ATen\native\TensorCompare.cpp:328.)
 score = torch.where(mask[i].unsqueeze(1), next_score, score)

**Comment by cutecharmingkid** [comment]
![image](https://user-images.githubusercontent.com/102043624/225285358-bf3b7081-182f-4d96-89b5-ad26b690986a.png)

**Comment by cutecharmingkid** [comment]
万分感激您的答疑解惑，还有一个问题我想咨询一下：当我在运行predict文件的时候，文件的第52行报错了，我想问下这里面被注释掉的drop_ratio换成什么比较好？





星辰大海
***@***.***



&nbsp;




------------------&nbsp;原始邮件&nbsp;------------------
发件人:                                                                                                                        "PeiChangY/-BERT-BILSTM-DGCNN-ATTENTION-CRF-"                                                                                    ***@***.***&gt;;
发送时间:&nbsp;2023年3月15日(星期三) 晚上6:29
***@***.***&gt;;
***@***.******@***.***&gt;;
主题:&nbsp;Re: [PeiChangY/-BERT-BILSTM-DGCNN-ATTENTION-CRF-] 关于训练完成后模型效果的提问 (Issue #3)





 建议您看一下我发表在知网上的论文《基于门控空洞卷积神经网络的中文命名实体识别》，里面讲的比较详细，有各个数据集的实体类型 
 
 
 
 
 ------------------&amp;nbsp;原始邮件&amp;nbsp;------------------ 
 发件人:                                                                                                                        "PeiChangY/-BERT-BILSTM-DGCNN-ATTENTION-CRF-"                                                                                    ***@***.***&amp;gt;; 
 发送时间:&amp;nbsp;2023年3月15日(星期三) 下午5:03 
 ***@***.***&amp;gt;; 
 ***@***.***&amp;gt;; 
 主题:&amp;nbsp;[PeiChangY/-BERT-BILSTM-DGCNN-ATTENTION-CRF-] 关于训练完成后模型效果的提问 (Issue #3) 
 
 
 
 
 
   
 您好，我是使用您的数据集msra_data进行训练，我使用predict文件预测使用的语句是：习近平是中华人民共和国的主席。返回的预测结果为空。同时系统提示：UserWarning: where received a uint8 condition tensor. This behavior is deprecated and will be removed in a future version of PyTorch. Use a boolean condition instead. (Triggered internally at ..\aten\src\ATen\native\TensorCompare.cpp:328.) 
  score = torch.where(mask[i].unsqueeze(1), next_score, score)

**Comment by we197064** [comment]
> 您好，我是使用您的数据集msra_data进行训练，我使用predict文件预测使用的语句是：习近平是中华人民共和国的主席。返回的预测结果为空。同时系统提示：UserWarning: where received a uint8 condition tensor. This behavior is deprecated and will be removed in a future version of PyTorch. Use a boolean condition instead. (Triggered internally at ..\aten\src\ATen\native\TensorCompare.cpp:328.) score = torch.where(mask[i].unsqueeze(1), next_score, score) ![887ea71b0140ffbaf084f56d77b478b](https://user-images.githubusercontent.com/102043624/225259519-c6a83f71-6601-42e6-b459-61c38851695a.png)
换成这个代码试试
![image](https://user-images.githubusercontent.com/48745533/225331053-2d4b6862-60e7-4c66-bdb1-e75bbb1dae8f.png)

**Comment by we197064** [comment]
> 万分感激您的答疑解惑，还有一个问题我想咨询一下：当我在运行predict文件的时候，文件的第52行报错了，我想问下这里面被注释掉的drop_ratio换成什么比较好？ 星辰大海 ***@***.*** &nbsp;
> […](#)
> ------------------&nbsp;原始邮件&nbsp;------------------ 发件人: "PeiChangY/-BERT-BILSTM-DGCNN-ATTENTION-CRF-" ***@***.***&gt;; 发送时间:&nbsp;2023年3月15日(星期三) 晚上6:29 ***@***.***&gt;; ***@***.******@***.***&gt;; 主题:&nbsp;Re: [PeiChangY/-BERT-BILSTM-DGCNN-ATTENTION-CRF-] 关于训练完成后模型效果的提问 (Issue #3) 建议您看一下我发表在知网上的论文《基于门控空洞卷积神经网络的中文命名实体识别》，里面讲的比较详细，有各个数据集的实体类型
> ------------------&amp;nbsp;原始邮件&amp;nbsp;------------------ 发件人: "PeiChangY/-BERT-BILSTM-DGCNN-ATTENTION-CRF-" ***@***.***&amp;gt;; 发送时间:&amp;nbsp;2023年3月15日(星期三) 下午5:03 ***@***.***&amp;gt;; ***@***.***&amp;gt;; 主题:&amp;nbsp;[PeiChangY/-BERT-BILSTM-DGCNN-ATTENTION-CRF-] 关于训练完成后模型效果的提问 (Issue #3) 您好，我是使用您的数据集msra_data进行训练，我使用predict文件预测使用的语句是：习近平是中华人民共和国的主席。返回的预测结果为空。同时系统提示：UserWarning: where received a uint8 condition tensor. This behavior is deprecated and will be removed in a future version of PyTorch. Use a boolean condition instead. (Triggered internally at ..\aten\src\ATen\native\TensorCompare.cpp:328.) score = torch.where(mask[i].unsqueeze(1), next_score, score) — Reply to this email directly, view it on GitHub, or unsubscribe. You are receiving this because you are subscribed to this thread.Message ID: ***@***.***&amp;gt; — Reply to this email directly, view it on GitHub, or unsubscribe. You are receiving this because you authored the thread.Message ID: ***@***.***&gt;
我添加了config.filters_number进去就没错误了

**Comment by cutecharmingkid** [comment]
> > 万分感激您的答疑解惑，还有一个问题我想咨询一下：当我在运行predict文件的时候，文件的第52行报错了，我想问下这里面被注释掉的drop_ratio换成什么比较好？ 星辰大海 _**@**_.***  
> > […](#)
> > ------------------ 原始邮件 ------------------ 发件人: "PeiChangY/-BERT-BILSTM-DGCNN-ATTENTION-CRF-" _**@**_._**>; 发送时间: 2023年3月15日(星期三) 晚上6:29 _**@**_.**_>; _**@**_._**_**@**_.**_>; 主题: Re: [PeiChangY/-BERT-BILSTM-DGCNN-ATTENTION-CRF-] 关于训练完成后模型效果的提问 (Issue #3) 建议您看一下我发表在知网上的论文《基于门控空洞卷积神经网络的中文命名实体识别》，里面讲的比较详细，有各个数据集的实体类型
> > ------------------&nbsp;原始邮件&nbsp;------------------ 发件人: "PeiChangY/-BERT-BILSTM-DGCNN-ATTENTION-CRF-" _**@**_._**&gt;; 发送时间:&nbsp;2023年3月15日(星期三) 下午5:03 _**@**_.**_&gt;; _**@**_._**&gt;; 主题:&nbsp;[PeiChangY/-BERT-BILSTM-DGCNN-ATTENTION-CRF-] 关于训练完成后模型效果的提问 (Issue #3) 您好，我是使用您的数据集msra_data进行训练，我使用predict文件预测使用的语句是：习近平是中华人民共和国的主席。返回的预测结果为空。同时系统提示：UserWarning: where received a uint8 condition tensor. This behavior is deprecated and will be removed in a future version of PyTorch. Use a boolean condition instead. (Triggered internally at ..\aten\src\ATen\native\TensorCompare.cpp:328.) score = torch.where(mask[i].unsqueeze(1), next_score, score) — Reply to this email directly, view it on GitHub, or unsubscribe. You are receiving this because you are subscribed to this thread.Message ID: _**@**_.**_&gt; — Reply to this email directly, view it on GitHub, or unsubscribe. You are receiving this because you authored the thread.Message ID: _**@**_.***>
> > 我添加了config.filters_number进去就没错误了

您好，我想问一下您模型训练完成之后的效果怎么样？能识别出人民/地名/组织名吗？

**Comment by cutecharmingkid** [comment]
> 建议您看一下我发表在知网上的论文《基于门控空洞卷积神经网络的中文命名实体识别》，里面讲的比较详细，有各个数据集的实体类型
> […](#)
> ------------------&nbsp;原始邮件&nbsp;------------------ 发件人: "PeiChangY/-BERT-BILSTM-DGCNN-ATTENTION-CRF-" ***@***.***&gt;; 发送时间:&nbsp;2023年3月15日(星期三) 下午5:03 ***@***.***&gt;; ***@***.***&gt;; 主题:&nbsp;[PeiChangY/-BERT-BILSTM-DGCNN-ATTENTION-CRF-] 关于训练完成后模型效果的提问 (Issue #3) 您好，我是使用您的数据集msra_data进行训练，我使用predict文件预测使用的语句是：习近平是中华人民共和国的主席。返回的预测结果为空。同时系统提示：UserWarning: where received a uint8 condition tensor. This behavior is deprecated and will be removed in a future version of PyTorch. Use a boolean condition instead. (Triggered internally at ..\aten\src\ATen\native\TensorCompare.cpp:328.) score = torch.where(mask[i].unsqueeze(1), next_score, score) — Reply to this email directly, view it on GitHub, or unsubscribe. You are receiving this because you are subscribed to this thread.Message ID: ***@***.***&gt;

作者您好，我想问一下，你使用这个训练好的模型去做预测的时候，预测产出的标签会不会不符合bio的标准?我今天下午做预测的时候发现有好几个实体直接是i开头的，不是b开头的。就比如说我预测一句话，北京是中国的首都。正常来说，北应该是B－LOC,但是结果预测成了I－LOC。我看模型里面有CRF层，应该是避免了这个问题的，但是实际上还是出现了标签预测不规范的问题
